src/components/Card/index.jsx

```diff
@@ -1,29 +1,88 @@
 import React, { Component } from 'react';
 import './index.css';
+import {BsPencil, BsX, BsCheck} from 'react-icons/bs';
 
 class Card extends Component {
     state = {
-        isChecked: false
+        isChecked: false,
+        isEdit: false,
+        cardData : {
+            Logo: 'Caption',
+            Text: 'Some text ...',
+        },
+        tempCardData: {
+            Logo: 'Caption',
+            Text: 'Some text ...'
+        }
     }
-
     changeStyles = () => {
         this.setState({isChecked :!this.state.isChecked});
     }
-
-    render(){
-        return(
+    changeEditStatus = () =>{
+        this.setState({isEdit : !this.state.isEdit, isChecked: false});
+    }
+    handleChangeInfo = (event,name) =>{
```

Comment:
As said before, you should copy cardData object to temp once switching to Edit mode, otherwise you'll only have initial values in temp object - and not resetted if you cancel them.